Add comments for exported symbols

kevgo opened the pull request:
The linter requires documentation for exported symbols. Documentation must start with the name of the exported symbol in Go. 

@charlierudolph

charlierudolph commented:
We export symbols in all the lib folders as well. Why aren't those needed? Maybe its only needed for the entry point, which may change based on #839?

kevgo commented:
I have added documentation to all exported symbols in `lib`. I also added fixes for other linter warnings here. Given that there are already open PRs for child branches, I could not separate them in a meaningful way without significant effort. The changes are very simple though.
